metricbeat/module/elasticsearch/ccr/ccr.go

```diff
@@ -107,3 +114,48 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 		return
 	}
 }
+
+func (m *MetricSet) checkCCRAvailability(currentElasticsearchVersion string) (message string, err error) {
+	license, err := elasticsearch.GetLicense(m.HTTP, m.getResetURI())
+	if err != nil {
+		return "", errors.Wrap(err, "error determining Elasticsearch license")
+	}
+
+	licenseType, err := license.GetValue("type")
+	if err != nil {
+		return "", errors.Wrap(err, "error determining Elasticsearch license type")
+	}
+
+	currentLicense, ok := licenseType.(string)
+	if !ok {
+		return "", fmt.Errorf("error determining Elasticsearch license type")
+	}
+
+	isAvailable := currentLicense == "trial" || currentLicense == "platinum"
+	if !isAvailable {
+		message = "the CCR feature is available with a platinum Elasticsearch license. " +
+			"You currently have a " + currentLicense + " license. " +
+			"Either upgrade your license or remove the ccr metricset from your Elasticsearch module configuration."
+		return
+	}
+
+	isAvailable, err = elastic.IsFeatureAvailable(currentElasticsearchVersion, elasticsearch.CCRStatsAPIAvailableVersion)
+	if err != nil {
+		return "", errors.Wrap(err, "error determining if CCR is available in current Elasticsearch version")
+	}
+
+	if !isAvailable {
+		metricsetName := m.FullyQualifiedName()
+		message = "the " + metricsetName + " is only supported with Elasticsearch >= " +
+			elasticsearch.CCRStatsAPIAvailableVersion + ". " +
+			"You are currently running Elasticsearch " + currentElasticsearchVersion + "."
+		return
+	}
+
+	return "", nil
+}
+
+func (m *MetricSet) getResetURI() string {
```

Comment:
Why reset?

Reply:
Ah, I was thinking of how this URI is being _used_, over here: https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/elasticsearch.go#L150

But in the context of the metricset alone, "reset" is not a good name. I'll change it to something like "service" instead.